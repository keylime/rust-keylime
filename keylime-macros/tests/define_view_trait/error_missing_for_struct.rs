use keylime_macros::define_view_trait;

#[define_view_trait]        // ← missing `for_struct = "..."`
struct BadView {
    x: u8,
}

fn main() {}
