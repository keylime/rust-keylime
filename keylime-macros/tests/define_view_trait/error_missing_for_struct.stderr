error: Usage: #[define_view_trait(for_struct = "MainStruct")]
 --> tests/define_view_trait/error_missing_for_struct.rs:3:1
  |
3 | #[define_view_trait]        // ‚Üê missing `for_struct = "..."`
  | ^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `define_view_trait` (in Nightly builds, run with -Z macro-backtrace for more info)
